samples/AzCopyCore/AzCopyCore/Program.cs

```diff
@@ -197,14 +197,27 @@ static void TransferStorageFileToDirectory(ReadOnlySpan<char> storageFile, ReadO
         {
             using (var bytes = new FileStream(localFilePath, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, true))
             {
-                var putRequest = new PutRangeRequest(storagePath, bytes);
-                response = await client.SendRequest(putRequest).ConfigureAwait(false);
-                if (response.StatusCode == 201) return true;
+                long bytesLeft = bytes.Length;
+                long index = 0;
+                int length = 1024 * 1024 * 4; 
+                while (true)
+                {
+                    if (bytesLeft < length) length = (int)bytesLeft;
+                    var putRequest = new PutRangeRequest(storagePath, bytes, index, length);
+                    response = await client.SendRequest(putRequest).ConfigureAwait(false);
+                    if (response.StatusCode != 201)
+                    {
+                        Log.TraceEvent(TraceEventType.Error, 0, "Response Status Code {0}", response.StatusCode);
+                        return false;
+                    }
+                    index += length;
+                    bytesLeft -= length;
+                    if (bytesLeft == 0) break;
```

Comment:
nit: why not use this as the while condition, especially since we do this check at the end of the loop anyway?

Reply:
will do